fauxfactory/__init__.py

```diff
@@ -520,25 +520,50 @@ def gen_negative_integer():
     return gen_integer(max_value=max_value)
 
 
-def gen_ipaddr(ip3=False, ipv6=False):
+def gen_ipaddr(ip3=False, ipv6=False, prefix=None):
     """Generates a random IP address.
 
+    You can also specify an IP address prefix if you are interested in
+    local network address generation, etc.
+
     :param bool ip3: Whether to generate a 3 or 4 group IP.
     :param bool ipv6: Whether to generate IPv6 or IPv4
+    :param list prefix: A prefix to be used for IPv4 (like [10, 0, 1])
     :returns: An IP address.
     :rtype: str
+    :raises: ``ValueError`` if ``prefix`` is too long either to fit into
+        the IP address or disabling randomness at all.
```

Comment:
@Ichimonji10 the `:raises:` cannot be so simply written because these limits also apply for ipv6 which has 8 fields therefore max. 7 fields of prefix.